Sources/Purchasing/StoreKitAbstractions/SK1StoreProductDiscount.swift

```diff
@@ -77,3 +79,20 @@ private extension StoreProductDiscount.PaymentMode {
     }
 
 }
+
+@available(iOS 11.2, macOS 10.13.2, tvOS 11.2, watchOS 6.2, *)
+private extension SK1ProductDiscount {
+
+    // See https://github.com/RevenueCat/purchases-ios/issues/1521
+    // Despite `SKProductDiscount.priceLocale` being non-optional, StoreKit might return `nil` `NSLocale`s.
+    // This works around that to make sure the SDK doesn't crash when bridging to `Locale`.
+    var optionalLocale: Locale? {
```

Comment:
the sadness. 

We've seen [these a few times before](https://github.com/RevenueCat/purchases-ios/blob/main/Sources/Purchasing/StoreKit1/StoreKitExtensions/SKProduct%2BExtensions.swift#L19), I wonder if we should start having like a file `StoreKit+NullabilityFixes`

Reply:
Actually I love that idea. `StoreKitWorkarounds`? To isolate the "real" code versus special cases for Apple bugs, like a certain other PDF company used to do with UIKit. That way we can keep track of these and hopefully deprecate them over time.

Reply:
❤️ it

Reply:
Done.